File: model/src/main/java/io/sparkled/model/validator/exception/EntityNotFoundException.java
package io.sparkled.model.validator.exception;

/**
 * Thrown when an entity lookup yields no result when there should definitely be one. The message contained in the
 * exception is guaranteed to be user friendly.
 */
public class EntityNotFoundException extends RuntimeException {

    public EntityNotFoundException(String message) {
        super(message);
    }

    public EntityNotFoundException(String message, Throwable cause) {
        super(message, cause);
    }
}

File: model/src/main/kotlin/io/sparkled/model/validator/exception/EntityNotFoundException.kt
package io.sparkled.model.validator.exception

/**
 * Thrown when an entity lookup yields no result when there should definitely be one. The message contained in the
 * exception is guaranteed to be user friendly.
 */
class EntityNotFoundException : RuntimeException {

    constructor(message: String) : super(message) {}

    constructor(message: String, cause: Throwable) : super(message, cause) {}
}
